media/src/app.ts

```diff
@@ -0,0 +1,276 @@
+import {config} from './config';
+import fs from 'fs';
+import crypto from 'crypto';
+import mysql, { RowDataPacket } from 'mysql2/promise';
+import sizeOf from 'image-size';
+import fetch from 'node-fetch';
+const http = require('http');
+
+const MEDIA_DOWNLOAD_REQUEST_TIMEOUT_MS = 120000;
+
+enum MediaType {
+  IMG = 'img',
+  VIDEO = 'video'
+}
+
+enum ItemType {
+  COMMENT = 'comment',
+  POST = 'post'
+}
+
+interface IContentSourceItem extends RowDataPacket {
+  content_source_id: number;
+  html: string;
+  ref_type: ItemType.COMMENT | ItemType.POST;
+  ref_id: number;
+}
+
+if (!fs.existsSync(config.dbFile)) {
```

Comment:
In the final version I suggest to replace files with a single `tinyint` flag in the `content_source`, similar to `indexed`.

However, it's worth to have a local cache of the processed urls.